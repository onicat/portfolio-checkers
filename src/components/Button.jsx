import React from 'react'

import 'styles/Button.css'

const Button = ({children, disabled, onClick}) => {
  return (
    <button 
      className={`Button ${(disabled) ? 'Button_disabled' : ''}`}
      onClick={(disabled) ? null : onClick}
    >
      {children}
    </button>
  )
};

export default Button;