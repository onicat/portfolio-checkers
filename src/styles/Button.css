.Button {
  margin: 0 2px;
  background-color: transparent;
  border: 1px solid #1D1D1D;
  border-radius: 5px;
}