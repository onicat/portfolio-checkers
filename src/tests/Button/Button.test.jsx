import React from 'react'
import TestRenderer from 'react-test-renderer';

import Button from 'components/Button';

describe('<Button />', () => {
  it('Render Button without className', () => {
    const tree = TestRenderer
      .create(<Button>Button</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot()
  })
});