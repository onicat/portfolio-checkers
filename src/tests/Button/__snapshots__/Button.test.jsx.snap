// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Render Button without className 1`] = `
<button
  className="Button "
>
  Button
</button>
`;
